nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -115,10 +116,10 @@ let
       jobClassLabel = toString ([""] ++ processCfg.jobClasses);
       threads = toString (if processCfg.threads == null then cfg.sidekiqThreads else processCfg.threads);
     in {
-      after = [ "network.target" "mastodon-init-dirs.service" ]
+      after = [ "network.target" "mastodon-init-dirs.service" "redis-mastodon.service" ]
```

Comment:
I can return check to `redis.createLocally` or remove this setting entirely.

Reply:
My proposal would be to remove all of `services.mastodon.redis.*` (and show appropriate error messages to users when they have set these options). This would probably also be best handled in a new PR. What do you think?

Reply:
Mh on the other hand, so many NixOS modules [seem to support external Redis hosts](https://search.nixos.org/options?channel=23.11&from=0&size=50&sort=relevance&type=packages&query=.redis.). Sorry for the back and forth, but the way you are currently going forward might be the one most aligned with other modules.

So in that case we'd only need to add redis-mastodon conditionally to the systemd dependencies.

Reply:
Let's leave extended support for redis.
I would like to bring back other advanced options as well.